opal/mca/common/ucx/common_ucx_wpool.c

```diff
@@ -670,7 +718,7 @@ _tlocal_mem_rec_cleanup(_mem_record_t *mem_rec)
     }
 
     opal_mutex_lock(&mem_rec->winfo->mutex);
```

Comment:
Move the code (lines 720-726) to the wpool scope and take wpool mutex instead of winfo mutex.